src/ixdat/techniques/deconvolution.py

```diff
@@ -1,217 +1,150 @@
+                will be calculated based on the measured data, overwriting the parameters
+                passed for a calculated one.
+            working_distance: Working distance between electrode and gas/liq interface in
+                m. Optional, though necessary if no data is provided.
+            A_el: Geometric electrode area in cm2. Optional, though necessary if no data
+                is provided.
+            diff_const: Diffusion constant in liquid. Optional. Default will check
+            diffusion constant in water in Molecule data from siq.
+            henry_vola: Dimensionless Henry volatility. Optional. Default will check
+            Henry volatility constant in water in Molecule data from siq.-CURRENTLY WRONG
+            chip: Optional. Needed to define capillary flow. Default will use
+                SpectroInlets chip from siq.
+            carrier_gas: The carrier gas used to calculate capillary flow. Defaults to He
+            gas_volume: the volume of the headspace volume in the chip. Default is the
+                volume of the SpectroInlets chip.
         """
-
-        if MS_data and parameters:  # TODO: Make two different classes
-            raise Exception(
-                "Kernel can only be initialized with data OR parameters, not both"
-            )
-        if EC_data and MS_data:
-            print("Generating kernel from measured data")
+        if data is not None:
+            self.mol = mol
+            self.data = data
             self.type = "measured"
-        elif parameters:
-            print("Generating kernel from parameters")
+            print("Generating `ECMSImpulseResponse` from measured data.")
+            if working_distance is not None or A_el is not None:
+                raise UserWarning(
+                    "Data was used to generate `ECMSImpulseResponse` ignoring the given"
+                    "working_distance/electrode area."
+                )
+
+        elif working_distance is not None and A_el is not None:
+            self.mol = mol
             self.type = "functional"
+            if not plugins.use_siq:
+                raise TechniqueError(
+                    "`ECMSImpulseResponse` will only work properly when using "
+                    # TODO this should be improved.- doesnt need to fully depend on siq
+                    # integration
+                    "`spectro_inlets_quantification` "
+                    "(`ixdat.plugins.activate_siq()`). "
+                )
+            # calculate the capillary flow for the specified gas & chip
+            Chip = plugins.siq.Chip
+            chip = chip or Chip()
+            n_dot = chip.calc_n_dot_0(gas=carrier_gas)
+            # convert to volumetric flux using T and p given by chip.
+            V_dot = n_dot * R * chip.T / chip.p
+            # find the other parameters from the siq Molecule files
+            Molecule = plugins.siq.Molecule
+            molecule = Molecule(mol)
```

Comment:
yeah, would make sense if there was an error message there. I figured this out in some other context later, but it is not super intuitive, since initiating a chip is done by `chip = Chip()`, isnt it. 
Anyway, not the place to discuss this here. Have changed it in the code

Reply:
There's also a `Chip.load()` method, which loads a chip from file. `Chip()` loads a standard chip, and can take parameters as input if they aren't standard. The problem is, there is no standard molecule. You could initiate a molecule by doing `Molecule(name="H2", molar_mass=2, D=...)`, but much easier to load it.